source/docs/software/wpilib-tools/shuffleboard/custom-widgets/creating-plugins.rst

```diff
@@ -0,0 +1,101 @@
+Creating a Plugin
+=================
+
+Overview
+--------
+Plugins provide the ability to create custom widgets, layouts, data sources/types, and custom themes. Shuffleboard provides the following plugins:
+
+- NetworkTables Plugin: To connect to data published over NetworkTables
+- Base Plugin: To display custom FRC data types in custom widgets
+- CameraServer Plugin: To view streams from the CameraServer
+
+Create a Custom Plugin
+----------------------
+In order to define a plugin, the plugin class must be a subclass of `edu.wpi.first.shuffleboard.api.Plugin <https://github.com/wpilibsuite/shuffleboard/blob/master/api/src/main/java/edu/wpi/first/shuffleboard/api/plugin/Plugin.java>`_ or one of its subclasses. An example of a plugin class would be as following.
+
+.. tabs::
+
+   .. code-tab:: java
+
+      import edu.wpi.first.shuffleboard.api.plugin.Description;
+      import edu.wpi.first.shuffleboard.api.plugin.Plugin;
+
+      @Description(group = "com.example", name = "MyPlugin", version = "1.2.3", summary = "An example plugin")
+      public class MyPlugin extends Plugin {
+
+      }
+
+More resources on how these attributes are used, including version numbers Shuffleboard's plugin dependencies can be found `here <https://semver.org/>`_.
```

Comment:
This feels incomplete. semver.org only describes the versioning scheme.

Reply:
The _group_, _name_, and _summary_ parameters seem quite self explanatory, versioning was specified since it must follow the systematic versioning guidelines.